#!/bin/bash

PATCH="firefox-unity-support.patch"
URELEASE="trusty"

patch_firefox() {
for i in www-client/firefox/firefox-${1}.*.ebuild; do
	cat > ${PATCH} << EOF
--- a/${i}	2013-10-12 22:06:13.218057063 +0200
+++ b/${i}	2013-10-12 22:37:42.203441496 +0200
@@ -44,7 +44,8 @@
 # More URIs appended below...
 SRC_URI="\${SRC_URI}
 	http://dev.gentoo.org/~anarchy/mozilla/patchsets/\${PATCH}.tar.xz
-	http://dev.gentoo.org/~nirbheek/mozilla/patchsets/\${PATCH}.tar.xz"
+	http://dev.gentoo.org/~nirbheek/mozilla/patchsets/\${PATCH}.tar.xz
+	https://launchpad.net/ubuntu/${URELEASE}/+source/\${PN}/${1}.0+build${2}/+files/\${PN}_${1}.0+build${2}.debian.tar.gz"
 
 ASM_DEPEND=">=dev-lang/yasm-1.1"
 
@@ -143,6 +144,7 @@
 }
 
 src_prepare() {
+	epatch "\${WORKDIR}/debian/patches/unity-menubar.patch"
 	# Apply our patches
 	EPATCH_SUFFIX="patch" \\
 	EPATCH_FORCE="yes" \\
EOF
	git apply ${PATCH}
	rm ${PATCH}
done
}

patch_firefox 24 1-0ubuntu1
patch_firefox 26 2-0ubuntu2
cat >> www-client/firefox/Manifest << EOF
DIST firefox_24.0+build1-0ubuntu1.debian.tar.gz 273652 SHA256 4e3f88622381def717b8f9d5c39aef916faadb88ced8c458b1496f187eadc390 SHA512 73dc738d2aab5f6dbc9faf60e61b0f1795db6ce773b13551207fc26a9ccd783a92cb0ca4f0c21f95e78b4ab74f9a62dc5d0497d2d92dca55f663d59e324b520d WHIRLPOOL 9f9aa0beb7fc69a97dd9381c88db5d62f9041dfed30f8470024f98a1a1d9cd0a8483e10ed018defd3c10673b84ddc55bdbe6d2e6049992264eeab818a36ad44e
DIST firefox_26.0+build2-0ubuntu2.debian.tar.gz 293562 SHA256 381b5531332eb952f334f85c5c05c5825aae84e0d4ece7af0c767c4d41d126c7 SHA512 38436d578ffb5ef83e24ccd4599e0ca4cc025b3a81f1b8aa7ae075e8b5e3daef69830dd0a6239d70e676710f245d63649907e54622672051bdd6ee62bf9f2c52 WHIRLPOOL c60b3fa9fbb78ec351719120d50df1277adc8b01e09402d1d9694b0578af2f7f8577bb07ccdece8b8ef1c4cafc9bb00802364d423d696543c5a05a12fbfeffda
EOF
